examples/api/lib/widgets/routes/flexible_route_transitions.0.dart

```diff
@@ -0,0 +1,226 @@
+                    builder: (BuildContext context) {
+                      return  const MyHomePage(title: 'Cupertino Transition');
+                    }
+                  );
+                  Navigator.of(context).push(route);
+                },
+                child: const Text('Cupertino Transition'),
+              ),
+            ],
+          ),
+        ),
+    );
+  }
+}
+
+class VerticalTransitionPageRoute<T> extends PageRoute<T> {
+  /// Creates a page route for use in an iOS designed app.
+  VerticalTransitionPageRoute({
+    required this.builder,
+  });
+
+  final WidgetBuilder builder;
+
+  @override
+  DelegatedTransitionBuilder? get delegatedTransition => VerticalPageTransition._delegatedTransition;
+
+  @override
+  Color? get barrierColor => const Color(0x00000000);
+
+  @override
+  bool get barrierDismissible => false;
+
+  @override
+  String? get barrierLabel => 'Should be no visible barrier...';
+
+  @override
+  bool get maintainState => true;
+
+  @override
+  bool get opaque => false;
+
+  // Begin PageRoute.
+
+  @override
+  Duration get transitionDuration => const Duration(milliseconds: 2000);
+
+  @override
+  Widget buildPage(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation) {
+    return builder(context);
+  }
+
+  static Widget buildPageTransitions<T>(
+    ModalRoute<T> route,
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  ) {
+    return VerticalPageTransition(
+      primaryRouteAnimation: animation,
+      secondaryRouteAnimation: secondaryAnimation,
+      child: child,
+    );
+  }
+
+  // TODO(justinmc): No canTransitionTo needed? I want to be able to have
+  // back-to-back VerticalPageTransitions.
+
+  @override
+  Widget buildTransitions(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget child) {
+    return buildPageTransitions<T>(this, context, animation, secondaryAnimation, child);
+  }
+
+  // End PageRoute.
+}
+
+class VerticalPageTransition extends StatelessWidget {
+  VerticalPageTransition({
+    super.key,
+    required Animation<double> primaryRouteAnimation,
+    required this.secondaryRouteAnimation,
+    required this.child,
+  }) : _primaryPositionAnimation =
+           CurvedAnimation(
+                 parent: primaryRouteAnimation,
+                 curve: curve,
+                 reverseCurve: curve,
+               ).drive(_kBottomUpTween),
+      _secondaryPositionAnimation =
+           CurvedAnimation(
+                 parent: secondaryRouteAnimation,
+                 curve: curve,
+                 reverseCurve: curve,
+               )
+           .drive(_kTopDownTween);
+
+  // When this page is coming in to cover another page.
+  final Animation<Offset> _primaryPositionAnimation;
+
+  // When this page is being coverd by another page.
```

Comment:
I misspelled "covered". Also it's up to you if these comments are necessary or not.